### PR TITLE
ENG-3675 template for OR-Tools

### PR DESCRIPTION
**Pull request by sebastian-quintero**
# Description

Enable the `ortools` python template.

Adds a workflow for Python linting with black and flake8.

### CODE REVIEW

File: .github/workflows/python-lint.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: git clone
+        uses: actions/checkout@v3
+
+      - name: set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+
+      - name: lint with black
+        uses: psf/black@stable
```
Comment (dirkschumacher) [inline .github/workflows/python-lint.yml]: Could you use a specific tag here vs the stable branch directly? E.g. [23.7.0](https://github.com/psf/black/releases/tag/23.7.0)

File: templates/ortools/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Nextmv OR-Tools Python template
+
+This template demonstrates how to solve a Mixed Integer Programming problem
+using the open-source open source software suite [OR-Tools][or-tools].
```
Comment (dirkschumacher) [inline templates/ortools/README.md]: Typo: "open-source open source"

File: templates/ortools/main.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    # Writes to stdout.
+    print(json.dumps(solution, indent=2))
+
+
+def solve(input_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    # Creates the solver.
+    provider = "SCIP"
```
Comment (dirkschumacher) [inline templates/ortools/main.py:26]: Is it possible to use highs here?

File: templates/ortools/main.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    # Solves the problem.
+    start_time = time.time()
+    status = solver.Solve()
+    end_time = time.time()
+
+    # Determines which items were chosen.
+    chosen_items = []
+    for item in items:
+        if item["variable"].solution_value() == 1.0:
```
Comment (dirkschumacher) [inline templates/ortools/main.py]: Do we need to take into account numerical problems. E.g. `0.9999` as the solution value? We could also do `item["variable"].solution_value() > 0.9` or something

**Review by dirkschumacher — CHANGES_REQUESTED** [review]
Some small requests/questions but looks good overall!

File: templates/ortools/main.py (hunk 3 above)
Comment (sebastian-quintero) [reply templates/ortools/main.py:26]: It is unsupported. 
image

File: templates/ortools/main.py (hunk 4 above)
Comment (sebastian-quintero) [reply templates/ortools/main.py]: Yep.

File: templates/ortools/README.md (hunk 2 above)
Comment (sebastian-quintero) [reply templates/ortools/README.md]: Thanks.

**Review by dirkschumacher — APPROVED** [review]

**Pull request merged**